celery/canvas.py

```diff
@@ -1123,18 +1123,17 @@ def set_immutable(self, immutable):
             task.set_immutable(immutable)
 
     def link(self, sig):
-        # Simply link to first task
+        # Simply link to first task. Doing this is slightly misleading because
+        # the callback may be executed before all children in the group are
+        # completed and also if any children other than the first one fail.
+        #
+        # The callback signature is cloned and made immutable since it the
+        # first task isn't actually capable of passing the return values of its
+        # siblings to the callback task.
         sig = sig.clone().set(immutable=True)
         return self.tasks[0].link(sig)
 
     def link_error(self, sig):
-        try:
```

Comment:
curious, removing this part won't have any negative effect?

Reply:
From what I can tell, no. I believe it was only cloned so that the `.set(immutable=True)` wouldn't alter the signature passed in by the user, and IMO making the errback immutable is probably the wrong thing to do. I explained (as I understand) the `.clone().set(immutable=True)` in the `link()` method above and since that explanation doesn't apply to the errbacks, I figured it made sense to remove it here.